File: src/main/java/net/adventuregame/guis/GuiShader.java
package net.adventuregame.guis;

import com.chidozie.core.shaders.ShaderProgram;
import org.joml.Matrix4f;

public class GuiShader extends ShaderProgram {

    private static final String VERTEX_FILE = "guis/guiVertexShader.vert";
    private static final String FRAGMENT_FILE = "guis/guiFragmentShader.frag";

    private int location_transformationMatrix;


    public GuiShader() {
        super(VERTEX_FILE, FRAGMENT_FILE);
    }

    public void loadTransformation(Matrix4f matrix){
        super.loadMatrix(location_transformationMatrix, matrix);
    }


    @Override
    protected void getAllUniformLocations() {
        location_transformationMatrix = super.getUniformLocation("transformationMatrix");
    }


    @Override
    protected void bindAttributes() {
        super.bindAttribute(0, "position");
    }
}

File: src/main/kotlin/net/adventuregame/guis/GuiShader.kt
package net.adventuregame.guis

import com.chidozie.core.shaders.ShaderProgram
import org.joml.Matrix4f

class GuiShader : ShaderProgram(VERTEX_FILE, FRAGMENT_FILE) {
    private var location_transformationMatrix = 0


    fun loadTransformation(matrix: Matrix4f) {
        super.loadMatrix(location_transformationMatrix, matrix)
    }


    override fun getAllUniformLocations() {
        location_transformationMatrix = super.getUniformLocation("transformationMatrix")
    }


    override fun bindAttributes() {
        super.bindAttribute(0, "position")
    }

    companion object {
        private const val VERTEX_FILE = "guis/guiVertexShader.vert"
        private const val FRAGMENT_FILE = "guis/guiFragmentShader.frag"
    }
}
